AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -128,6 +145,52 @@ - (NSArray *)children
   return self.layoutChildren[kDefaultChildrenKey];
 }
 
+
+#pragma mark - ASEnvironment
+
+- (ASEnvironmentCollection *)environmentCollection
```

Comment:
Move this as instance variable and implement the defines in ASLayoutSpec and ASDisplayNode